autoload/probe/file.vim

```diff
@@ -72,6 +76,26 @@ function! probe#file#scan()
     return s:file_caches[s:hash]
 endfunction
 
+function! s:is_git()
+    let metadir_name = fnamemodify(s:find_metadir(), ':t')
+    return metadir_name == '.git'
+endfunction
+
+function! s:scan_git()
+    let files = []
+    echo "Scanning " . s:find_repo_root()
+
+    let cmd = "git ls-files; git submodule foreach -q 'for file in $(git ls-files); do echo \"$path/$file\"; done'"
```

Comment:
Ah OK, it turns out that I just misunderstood how the flags in `git ls-files` actually worked.

It seems like `git ls-files --cached --others --exclude-standard` works fine (although I might need to add another option to ignore untracked at some point...).